### PR TITLE
Add dns resolution on endpoint parsing

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

This diff adds the option to do DNS resolution on registering replicas and/or coordinator instances.

Fixes: #1994 

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - Add DNS resolution on endpoint parsing

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug/feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - DNS resolution now works for endpoints
- [x] Link the documentation PR here
    - **[Documentation PR link]**
- [x] Tag someone from the docs team in the comments

### CODE REVIEW

**Comment by antoniofilipovic** [comment]
@kgolubic tagging you here

File: src/io/network/endpoint.cpp [hunk 1]
```diff
@@ -9,80 +9,105 @@
...
-    : address(std::move(hostname)), port(port), family{GetIpFamily(address)} {}
-
-Endpoint::Endpoint(std::string ip_address, uint16_t port) : address(std::move(ip_address)), port(port) {
-  IpFamily ip_family = GetIpFamily(address);
-  if (ip_family == IpFamily::NONE) {
-    throw NetworkError("Not a valid IPv4 or IPv6 address: {}", ip_address);
+Endpoint::Endpoint(std::string const &ip_address, uint16_t port) {
+  auto result = TryResolveAddress(ip_address, port);
+  if (!result.has_value()) {
+    throw NetworkError("Couldn't resolve neither DNS, nor IP address");
```
Comment (as51340) [inline src/io/network/endpoint.cpp:47]:
```suggestion
    throw NetworkError("Couldn't resolve neither using DNS, nor IP address");
```

**Review by as51340 — APPROVED** [review]
LGTM  :muscle: 

One suggestion though about exception message

**Comment by kgolubic** [comment]
@antoniofilipovic, can you please provide me with a Release note? What was done, and what can users do that they couldn't do before (since this is tagged as a feature)?

**Pull request merged**